### PR TITLE
Properties with mysql custom "columnName" don't get autoupdated

### PR DESCRIPTION
**Pull request by darknos**
### Description

#250 for 3.x

connect to #250 

### Checklist

- [+] New tests added or existing tests modified to cover all changes
- [+] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

File: lib/migration.js [hunk 1]
```diff
@@ -186,7 +186,7 @@ function mixinMigration(MySQL, mysql) {
     function actualize(propName, oldSettings) {
       var newSettings = m.properties[propName];
       if (newSettings && changed(newSettings, oldSettings)) {
-        var pName = self.client.escapeId(propName);
+        var pName = self.columnEscaped(model, propName);
```
Comment (ssh24) [inline lib/migration.js]: change this to `var pName = (newSettings.mysql && newSettings.mysql.columnName) || self.client.escapeId(propName);`

File: lib/migration.js [hunk 2]
```diff
@@ -176,7 +176,7 @@ function mixinMigration(MySQL, mysql) {
         });
       }
       if (found) {
-        actualize(colName, found);
+        actualize(propName, found);
```
Comment (ssh24) [inline lib/migration.js]: Why do we need change this?

Comment (darknos) [inline lib/migration.js]: because we have a way how to find column name by property name but not vise versa.

File: lib/migration.js (hunk 1 above)
Comment (darknos) [inline lib/migration.js]: but in this case columnName will be not escaped and we have possible issue with columnName like `from`.

Comment (darknos) [inline lib/migration.js]: I want to merge this PR asap, but why do you want to replace documented and widely used `self.columnEscaped` to that.

... ten lines above we use `var colName = expectedColNameForModel(propName, m);` for the same result. 

please, confirm the change and, please, explain the reason

Comment (ssh24) [inline lib/migration.js]: Right. But what I am not understanding is, if we specify the `columnName` explicitly to be something different than the property name, how would this work?

Comment (ssh24) [inline lib/migration.js]: Something like this?
```
var colName = newSettings.mysql && newSettings.mysql.columnName || self.client.escapeId(propName);
var pName = self.columnEscaped(model, colName);
```

Comment (darknos) [inline lib/migration.js]:
```javascript
var pName = self.client.escapeId(newSettings.mysql && newSettings.mysql.columnName || propName);
```
do the same as
```javascript
var pName = self.columnEscaped(propName);
```


```javascript
/**
 * Get the escaped column name for a given model property
 * @param {String} model The model name
 * @param {String} property The property name
 * @returns {String} The escaped column name
 */
SQLConnector.prototype.columnEscaped = function(model, property) {
  return this.escapeName(this.column(model, property));
};
```
```javascript
/**
 * Get the column name for the given model property. The column name can be
 * customized at the model property definition level as `column` or
 * `columnName`. For example,
 *
 * ```json
 * "name": {
 *   "type": "string",
 *   "mysql": {
 *     "column": "NAME"
 *   }
 * }
 * ```
 *
 * @param {String} model The model name
 * @param {String} property The property name
 * @returns {String} The column name
 */
SQLConnector.prototype.column = function(model, property) {
  var prop = this.getPropertyDefinition(model, property);
  var columnName;
  if (prop && prop[this.name]) {
    columnName = prop[this.name].column || prop[this.name].columnName;
    if (columnName) {
      // Explicit column name, return as-is
      return columnName;
    }
  }
  columnName = property;
  if (typeof this.dbName === 'function') {
    columnName = this.dbName(columnName);
  }
  return columnName;
};
```

**Comment by ssh24** [comment]
@slnode ok to test

**Comment by ssh24** [comment]
@slnode test please

**Comment by ssh24** [comment]
@darknos Thank you for taking the initiative and submitting this patch.

**Pull request merged**